src/metrics_server.rs

```diff
@@ -0,0 +1,171 @@
+    )
+    .unwrap();
+}
+
+enum ProbeResult {
+    IndexerStarted,
+    IndexerNotStarted,
+    SomethingIsWrong,
+}
+
+async fn serve_req(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    let response = match (req.method(), req.uri().path()) {
+        (&Method::GET, "/metrics") => {
+            let encoder = prometheus::TextEncoder::new();
+
+            let mut buffer = Vec::new();
+            if let Err(e) = encoder.encode(&prometheus::gather(), &mut buffer) {
+                eprintln!("could not encode custom metrics: {}", e);
+            };
+
+            let res = match String::from_utf8(buffer.clone()) {
+                Ok(v) => v,
+                Err(e) => {
+                    eprintln!("custom metrics could not be from_utf8'd: {}", e);
+                    String::default()
+                }
+            };
+            buffer.clear();
+
+            Response::builder()
+                .status(200)
+                .header(CONTENT_TYPE, encoder.format_type())
+                .body(Body::from(res))
+                .unwrap()
+        }
+        (&Method::GET, "/probe") => {
+            // Indexer should process at least 1 block within 2.5 minutes to report that it is working fine.
+            let encoder = prometheus::TextEncoder::new();
+            let mut probe_result = ProbeResult::IndexerNotStarted;
+            let latest_block_timestamp = LATEST_BLOCK_TIMESTAMP.get();
+            let last_known_blockheight = LATEST_BLOCK_HEIGHT.get() as u64;
+
+            let time_since_the_epoch = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards");
+            let time_elapsed = time_since_the_epoch
+                .checked_sub(Duration::from_secs(latest_block_timestamp as u64))
+                .unwrap();
+
+            if latest_block_timestamp != 0.0 {
+                match time_elapsed.as_secs().ge(&Duration::new(150, 0).as_secs()) {
+                    true => probe_result = ProbeResult::SomethingIsWrong,
+                    false => probe_result = ProbeResult::IndexerStarted,
+                }
+            }
+            let mut res = "".to_owned();
+
+            match probe_result {
+                ProbeResult::IndexerNotStarted => res.push_str("Indexer has not started yet."),
+                ProbeResult::IndexerStarted => {
+                    res.push_str("Indexer is operating normally. Blocks reported at least every 2.5 minutes.");
+                }
+                ProbeResult::SomethingIsWrong => {
+                    res.push_str("Something is wrong. Indexer has not reported any new blocks within 2.5 minutes.");
+                }
+            };
+
+            if latest_block_timestamp != 0.0 {
+                res.push_str("\n Last block height was: ");
+                res.push_str(last_known_blockheight.to_string().as_str());
+                res.push_str("\n last block was indexed at timestamp: ");
```

Comment:
I prefer to see the duration between the block timestamp and now. 
With just the block_timestamp, we force the developer to remember the diff between their timezone and the server timezone